docs/source/our_stack.rst

```diff
@@ -10,31 +10,108 @@ The most up-to-date reference on our codebase is our code itself.
     To see when any documentation page was last updated, click "Edit on GitHub" in
     the top right, then see the timestamp on the last commit.
 
-...
-TODO(Kevin): stub out the rest
-
-Overview
+Gameplay
 --------
+Gameplay is where our strategy is produced. It takes in the current state and
+assigns all robots a task based on some coordination. Currently, we have a
+system with a play selector that chooses a play based on world state. Then
+that play assigns roles, roles select a tactic based on their progress, and
+tactic select an atomic skill to perform. The gameplay loop is set to tick at
+60Hz. The below code block is how currently that is done. Each time the timer
+ends, the callback method is entered:
 
-ROS2
-----
+.. code-block:: python
 
-Vision
-------
+        timer_period = 1 / 60  # seconds
+        self.timer = self.create_timer(timer_period, self.gameplay_tick)
 
-Gameplay
---------
+Motion Control
+--------------
+A lot of the motion control is done on robot by the firmware, but part of the
+motion control is handled by software. See the ``control`` directory,
+``motion_control.cpp`` uses PID to correct robot position. The PID
+controller implementation is not in software, but it is in the
+`fshare repository <https://github.com/RoboJackets/robocup-fshare>`_.
+Correction of robot velocity is done by robot firmware (the code that runs on
+each robot's microcontroller) in the separate firmware repository.
 
 Motion Planning
 ---------------
+Motion planning is about generating a path from a robots current position and
+orientation (pose) to a goal supplied by gameplay. Starting at the lowest
+level, we use RRT for path generation. This occurs after trying to apply a
+straight line path to the goal and that is unsuccessful (if there are no
+obstacles in that straight line why bother with RRT?). We have our own RRT
+implementation in `a separate repository <https://github
+.com/RoboJackets/rrt>`_ which we use as a submodule. This
+way others can easily use our RRT implementation if they want.
```

Comment:
not just robocup teams can use that repo and it was actually a relatively popular project among random people. It is rather old now and could use a facelift though. Especially since we still use it.

Reply:
I'm not a big fan of the coding style employed in that repo. I feel someone would need to clean it up before we can add new features to it again.